src/transformers/modeling_flaubert.py

```diff
@@ -162,11 +162,13 @@ def forward(
         else:
             bs, slen = inputs_embeds.size()[:-1]
 
+        device = input_ids.device if input_ids is not None else inputs_embeds.device
+
         if lengths is None:
             if input_ids is not None:
                 lengths = (input_ids != self.pad_index).sum(dim=1).long()
             else:
-                lengths = torch.LongTensor([slen] * bs)
+                lengths = torch.tensor([slen] * bs, device=device)
```

Comment:
Why not LongTensor?

Reply:
LongTensor isn't a factory to create tensors, but the tensor class, so you can't directly set the device type on it. You would have to do:

```py
lengths = torch.LongTensor([slen] * bs).to(device)
```

If I understand correctly, this first creates the tensor on the CPU, before creating a new one on the updated device (if the device is different). When using the factory method, I believe it's only created once, and is therefore more efficient. I didn't check the source code directly though so I may be wrong.

Reply:
Makes sense, thanks for clarifying!